docs/pages/guides/testing-rsc.mdx

```diff
@@ -0,0 +1,119 @@
+
+### Setup
+
+While standard server rendering is web-only, Expo's universal React Server Components bundle custom server renderers for each platform. This means platform-extensions like `*.ios.js` and `*.native.ts` will be resolved when writing server components for an iOS app.
+
+`jest-expo` provides a couple different presets for testing server components:
+
+- `jest-expo/rsc/web`: An Android-only runner for React Server Components. Uses `*.web.js` and `*.js` files.
+- `jest-expo/rsc/ios`: An iOS-only runner for React Server Components. Uses `*.ios.js`, `*.native.js`, and `*.js` files.
+- `jest-expo/rsc/android`: An Android-only runner for React Server Components. Uses `*.android.js`, `*.native.js`, and `*.js` files.
+- `jest-expo/rsc`: A multi-runner which combines the above runners.
+
+We recommend creating a `jest-rsc.config.js` file in your project root to configure Jest for React Server Components:
+
+```js jest-rsc.config.js
+module.exports = require('jest-expo/rsc/jest-preset');
+```
+
+Then you can add a script like `test:rsc` to your `package.json`:
+
+```json package.json
+{
+  "scripts": {
+    "test:rsc": "jest --config jest-rsc.config.js"
+  }
+}
+```
+
+### Writing tests
+
+Tests should be written in a `__rsc_tests__` directory to prevent Jest from running your client tests on the server.
+
+```js __rsc_tests__/my-component.test.ts
+/// <reference types="jest-expo/rsc/expect" />
+
+import { LinearGradient } from 'expo-linear-gradient';
+
+it(`renders to RSC`, async () => {
+  const jsx = (
+    <LinearGradient
+      colors={['cyan', '#ff00ff', 'rgba(0,0,0,0)', 'rgba(0,255,255,0.5)']}
+      testID="gradient"
+    />
+  );
+
+  await expect(jsx).toMatchFlight(`1:I["src/LinearGradient.tsx",[],"LinearGradient"]
+0:["$","$L1",null,{"colors":["cyan","#ff00ff","rgba(0,0,0,0)","rgba(0,255,255,0.5)"],"testID":"gradient"},null]`);
+});
+```
+
+Any code you import in your tests will be run in the server environment, so you can import server-only modules like `react-server` and `server-only`. This is useful for determining if a library is compatible with React Server Components.
+
+### Custom expect matchers
+
+Jest Expo for RSC adds a couple of custom matchers to Jest's expect:
+
+- `toMatchFlight`: Render a JSX element using a pseudo-implementation of the render in Expo CLI and compare to a flight string.
+- `toMatchFlightSnapshot`: Same as `toMatchFlight` but saves the flight string to a snapshot file.
+
+Behind the scenes, these methods are performing part of the framework operation required to render React Server Components. The component is streamed to a string and compared all at once. You can alternatively stream it manually to observe the progress.
+
+If a component fails to render, the matcher will throw an error to fail the test. In reality, an `E:` line will be present in your flight string, this will be sent to the client where the error will end up being thrown.
```

Comment:
```suggestion
If a component fails to render, the matcher will throw an error to fail the test. An `E:` line will be present in your flight string in real time. This will be sent to the client, where the error will be thrown.
```